platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/type/ProjectTypeRegistry.java

```diff
@@ -1,167 +1,258 @@
-/*******************************************************************************
- * Copyright (c) 2012-2015 Codenvy, S.A.
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License v1.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-v10.html
- *
- * Contributors:
- *   Codenvy, S.A. - initial API and implementation
- *******************************************************************************/
 package org.eclipse.che.api.project.server.type;
 
+import org.eclipse.che.api.core.NotFoundException;
 import org.eclipse.che.api.core.model.project.type.Attribute;
-import org.eclipse.che.api.core.model.project.type.ProjectType;
 import org.eclipse.che.api.project.server.ProjectTypeConstraintException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Comparator;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 import java.util.regex.Pattern;
 
 /**
+ * Registry for Project Type definitions on server
+ *
+ * All the Project Types definitions should be registered here
+ *
  * @author gazarenkov
  */
 @Singleton
 public class ProjectTypeRegistry {
 
-    private static final Logger LOG = LoggerFactory.getLogger(ProjectTypeRegistry.class);
+    protected static final Logger LOG = LoggerFactory.getLogger(ProjectTypeRegistry.class);
```

Comment:
I don't see why the logger would be shared at protected level as it's using the class name as scope